### PR TITLE
[TIMOB-23828] iOS 10: Expose the CSSearchQuery API

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23828

### CODE REVIEW

File: apidoc/Titanium/App/iOS/SearchQuery.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    properties:
+      - name: success
+        summary:  Indicates if the operation succeeded. Returns true if download succeeded, false otherwise.
+        type: Boolean
+      - name: error
+        summary: Error message, if any returned. Undefined otherwise.
+        type: String
+
+examples:
+  - title: Perform a simple search-query for all items at include a "Searchable" in it.
```
Comment (cheekiatng) [inline apidoc/Titanium/App/iOS/SearchQuery.yml]: change 'at' with 'that'

File: iphone/Classes/TiAppiOSSearchQueryProxy.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2016 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+#ifdef IS_XCODE_8
```
Comment (cheekiatng) [inline iphone/Classes/TiAppiOSSearchQueryProxy.h]: `#if` not `#ifdef`

File: iphone/Classes/TiAppiOSSearchQueryProxy.m [hunk 3]
```diff
@@ -0,0 +1,101 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2016 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+#ifdef IS_XCODE_8
```
Comment (cheekiatng) [inline iphone/Classes/TiAppiOSSearchQueryProxy.m]: `#if` not `#ifdef`

File: iphone/Classes/TiAppiOSProxy.m [hunk 4]
```diff
@@ -20,6 +20,12 @@
 #import "TiAppiOSSearchableItemProxy.h"
 #import "TiAppiOSSearchableIndexProxy.h"
 
+#ifdef IS_XCODE_8
```
Comment (cheekiatng) [inline iphone/Classes/TiAppiOSProxy.m]: `#if` not `#ifdef`

File: iphone/Classes/TiAppiOSProxy.m [hunk 5]
```diff
@@ -250,6 +254,27 @@ -(id)createSearchableItemAttributeSet:(id)args
 }
 #endif
 
+#ifdef IS_XCODE_8
```
Comment (cheekiatng) [inline iphone/Classes/TiAppiOSProxy.m]: `#if` not `#ifdef`

**Comment by cheekiatng** [comment]
Please address comments. And we like this for 5.5.0 so please change docs min version to 5.5.0 as well.

**Comment by hansemannn** [comment]
Addressed.

**Comment by cheekiatng** [comment]
APPROVED!

**Pull request merged**